packages/puppeteer-core/src/bidi/core/UserContext.ts

```diff
@@ -0,0 +1,113 @@
+import type * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
+
+import {EventEmitter} from '../../common/EventEmitter.js';
+import {assert} from '../../util/assert.js';
+
+import type {Browser} from './Browser.js';
+import {BrowsingContext} from './BrowsingContext.js';
+
+export type CreateBrowsingContextOptions = Omit<
+  Bidi.BrowsingContext.CreateParameters,
+  'type' | 'referenceContext'
+> & {
+  referenceContext?: BrowsingContext;
+};
+
+export class UserContext extends EventEmitter<{
```

Comment:
I've implemented the temporary spec here. Do you think we shouldn't add this yet? I'm quite certain that it will be added and this class simplifies our implementation.